### DISCUSSION TITLE
repo.saltproject.io

### DISCUSSION THREAD

**Discussion by its-a-me-mario-the-og** [Security-Onion-Solutions/securityonion#13918] [2.4]
### Version

2.4.110

### Installation Method

Network installation on Debian (unsupported)

### Description

upgrading

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

12

### RAM

64

### Storage for /

1 TB

### Storage for /nsm

4.5 TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

Hello Security Onion Team and Community!

Wednesday I had executed soup on one of my managers, ran the soup command twice, after the second run of the command it failed to up date. This is a distributed deployment and the reason it failed is because it could not find repo.saltproject.io url because they are in the process of migrating to Broadcom.

After hours of trying to fix this, I finally have it fixed. After making a few changes to code in /securityonion/setup/so-functions and making changes to lines 1885 & 1886 to reflect this article:

https://saltproject.io/blog/salt-project-package-repo-migration-and-guidance/

and also made changes to lines 1893 & 1894 to reflect 1885 & 1886 in so-function.

Also had to make changes in /securityonion/salt/salt/scripts/bootstrap-salt.sh at line 9014 to reflect the article above. Once I did all of that the apt-get update worked fine. But once I tried to do ./so-setup-network for the forward sensors to communicate with the manager I got problems with salt downgrading.

Once I ran 
```
echo 'Package: salt-*
Pin: version 3006.*
Pin-Priority: 1001' | sudo tee `/etc/apt/preferences.d/salt-pin-1001
```

in the /etc/apt/preferences.d/

and ran sudo apt-get install salt-minion

and run ./so-setup-network once again everything worked fine after that.

I wanted to make the team and community aware of this and if anyone runs into the same problem to hopefully help them out as well. Hope this gets fixed soon! Thank you!

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
https://github.com/Security-Onion-Solutions/securityonion/issues/13926 There is an issue for this.

**Comment by GreatKampfstern** [comment]
Thank you, because the version with the fixed issue isn't out yet your fix is really helpful.
Could you assist me at the following question: You said you changed line 9014 at bootstrap-salt.sh to reflect the article but what do you mean by that? The line shows a url to a json, but in the article from salt project is no url provided for a json download and I tried to replace the "repo.saltproject.io" in the url with "packages.broadcom.com" like the article said, but this wasn't right.
Thank you in advance!

**Reply by its-a-me-mario-the-og** [reply]
I got you, go back to the article and find the distro you are using, the one I'm using is Debian so I used the Debian instructions.

What I did for bootstrap-salt.sh is I put "https://packages.broadcom.com/artifactory/api/security/keypair/SaltProjectKey/public" it's under where it says # Download public key in the article

I would also like to add it is probably best to go into /etc/apt/sources.list.d/ and remove the salt.list file, because if you don't remove it and try apt-get update it will still try to pull in from repo.saltproject.io 

I hope this helps you some more!
